sdk/tests/core/CMakeLists.txt

```diff
@@ -3,11 +3,111 @@
 
 cmake_minimum_required (VERSION 3.10)
 
-project (az_core_test LANGUAGES C)
+project (az_core_test_lib_test LANGUAGES C)
 
 set(CMAKE_C_STANDARD 99)
 
 include(AddCMockaTest)
+include(CheckAndIncludeCodeCov)
+
+# Define path to core
+set(AZ_CORE_PATH ${CMAKE_SOURCE_DIR}/sdk/src/azure/core)
+
+if (AZ_PLATFORM_IMPL STREQUAL "POSIX" AND NOT APPLE)
+  add_library (
+    az_core_test_lib
+    ${AZ_CORE_PATH}/az_base64.c
+    ${AZ_CORE_PATH}/az_base64.c
+    ${AZ_CORE_PATH}/az_context.c
+    ${AZ_CORE_PATH}/az_hfsm.c
+    ${AZ_CORE_PATH}/az_event_pipeline.c
+    ${AZ_CORE_PATH}/az_event_policy_collection.c
+    ${AZ_CORE_PATH}/az_http_pipeline.c
+    ${AZ_CORE_PATH}/az_http_policy.c
+    ${AZ_CORE_PATH}/az_http_policy_logging.c
+    ${AZ_CORE_PATH}/az_http_policy_retry.c
+    ${AZ_CORE_PATH}/az_http_request.c
+    ${AZ_CORE_PATH}/az_http_response.c
+    ${AZ_CORE_PATH}/az_json_reader.c
+    ${AZ_CORE_PATH}/az_json_token.c
+    ${AZ_CORE_PATH}/az_json_writer.c
+    ${AZ_CORE_PATH}/az_log.c
+    ${AZ_CORE_PATH}/az_mqtt5_connection.c
+    ${AZ_CORE_PATH}/az_mqtt5_connection_hfsm.c
+    ${AZ_CORE_PATH}/az_mqtt5_rpc_server_hfsm.c
+    ${AZ_CORE_PATH}/az_mqtt5_rpc_server_codec.c
+    ${AZ_CORE_PATH}/az_mqtt5_rpc_client_hfsm.c
+    ${AZ_CORE_PATH}/az_mqtt5_rpc_client_codec.c
+    ${AZ_CORE_PATH}/az_mqtt5_telemetry_consumer_hfsm.c
+    ${AZ_CORE_PATH}/az_mqtt5_telemetry_consumer_codec.c
+    ${AZ_CORE_PATH}/az_mqtt5_telemetry_producer_hfsm.c
+    ${AZ_CORE_PATH}/az_mqtt5_telemetry_producer_codec.c
+    ${AZ_CORE_PATH}/az_mqtt5_topic_parser.c
+    ${AZ_CORE_PATH}/az_precondition.c
+    ${AZ_CORE_PATH}/az_span.c
+  )
+else()
+  add_library (
+    az_core_test_lib
+    ${AZ_CORE_PATH}/az_base64.c
+    ${AZ_CORE_PATH}/az_context.c
+    ${AZ_CORE_PATH}/az_http_pipeline.c
+    ${AZ_CORE_PATH}/az_http_policy.c
+    ${AZ_CORE_PATH}/az_http_policy_logging.c
+    ${AZ_CORE_PATH}/az_http_policy_retry.c
+    ${AZ_CORE_PATH}/az_http_request.c
+    ${AZ_CORE_PATH}/az_http_response.c
+    ${AZ_CORE_PATH}/az_json_reader.c
+    ${AZ_CORE_PATH}/az_json_token.c
+    ${AZ_CORE_PATH}/az_json_writer.c
+    ${AZ_CORE_PATH}/az_log.c
+    ${AZ_CORE_PATH}/az_precondition.c
+    ${AZ_CORE_PATH}/az_span.c
+  )
+endif()
+
+target_include_directories(az_core_test_lib 
+  PUBLIC 
+  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/sdk/inc>
+  $<INSTALL_INTERFACE:include/az_core_test_lib_test>
+)
+
+
+# Custom options for the core tests
+add_compile_definitions(AZ_MQTT5_USER_CONFIG)
```

Comment:
Does the AZ_MQTT5_USER_CONFIG_PATH need to be set here as well?

Reply:
Not here, the configuration file for test is in the include folder for tests